code/game/machinery/ARES/ARES_procs.dm

```diff
@@ -0,0 +1,310 @@
+GLOBAL_DATUM_INIT(ares_link, /datum/ares_link, new)
+
+/datum/ares_link
+	var/link_id = MAIN_SHIP_DEFAULT_NAME
+	var/list/linked_systems = list()
+	var/obj/structure/machinery/ares/processor/interface/p_interface
+	var/obj/structure/machinery/ares/processor/apollo/p_apollo
+	var/obj/structure/machinery/ares/processor/bioscan/p_bioscan
+	var/obj/structure/machinery/computer/ares_console/interface
+
+/datum/ares_link/Destroy()
+	for(var/obj/structure/machinery/ares/link in linked_systems)
+		link.delink()
+	for(var/obj/structure/machinery/computer/ares_console/interface in linked_systems)
+		interface.delink()
+	..()
+
+/proc/log_apollo(speaker, message)
+	if(!speaker)
+		speaker = "Unknown"
+	var/datum/ares_link/link = GLOB.ares_link
+	if(!link.p_apollo || link.p_apollo.inoperable())
+		return
+	if(!link.p_interface || link.p_interface.inoperable())
+		return
+	if(!link.interface || link.interface.inoperable())
+		return
+	else
+		link.interface.apollo_log.Add("[worldtime2text()]: [speaker], '[message]'")
+
+/obj/structure/machinery/computer/ares_console/proc/get_ares_access(obj/item/card/id/card)
+	if(777 in card.access)
+		return ARES_ACCESS_DEBUG
```

Comment:
777 is a magic number, change it to a define

Reply:
It's deliberately a magic number because as of the moment I don't intend to keep it when the PR is finished.

Reply:
Make sure to include a comment saying that it's temporary so that it doesn't accidentally get merged in review

Reply:
There is one, it's just not in that specific place.
```
/// Debugging. Allows me to view everything without using a high command rank. Unlikely to stay in a full merge.
#define ARES_ACCESS_DEBUG 10
```

Reply:
always make sure to mark these things with a comment like `TODO: Remove`
Before merging code, I'll usually search for specific keywords like `TODO` to make sure the author hasn't forgotten to remove any comments